src/datastore/rbac-policies.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <set>
+#include <string>
+
+#include "pg.h"
+
+namespace datastore {
+class RbacPolicy {
+public:
+	using principal_t  = std::string;
+	using principals_t = std::set<principal_t>;
+	using role_t       = std::string;
+	using roles_t      = std::set<role_t>;
```

Comment:
assuming something like this works:
```cpp
const RbacPolicy::Principals RbacPolicy::principals() const {
	std::string_view qry = R"(
		select
			identity_id
		from
			"rbac-policies_identities"
		where
			policy_id = $1::text;
	)";

	auto res = pg::exec(qry, id());

	Principals members;
	for (const auto &r : res) {
		members.push_back(Principal({
			.id = r["identity_id"].as<std::string>(),
		}));
	}

	return members;
}
```

Reply:
I don't mind using structs.

Looking at the PR it seems `principal` is assumed to be an identity, which isn't the case. A principal can be either a collection or an identity. 

Instead of having the `type` in _Principal_ struct (which should be an enum instead of a struct if we were to keep it), [`std::variant`](https://en.cppreference.com/w/cpp/utility/variant) might be another option.